src/Discord.Net.SlashCommands/SlashModuleBase.cs

```diff
@@ -0,0 +1,54 @@
+using System;
+using System.Threading.Tasks;
+
+namespace Discord.SlashCommands
+{
+    /// <summary>
+    /// Base class for any Slash command handling modules
+    /// </summary>
+    /// <typeparam name="T">Type of slash command context to be injected into the module</typeparam>
+    public abstract class SlashModuleBase<T> : ISlashModuleBase where T : class, ISlashCommandContext
```

Comment:
I feel like the executing `RestApplicationCommand` should be added to the context to allow the user to interact/modify the current command.

Reply:
fetching the Application Command everytime a command is executed will really slow down the command execution operation. And i dont think the bot caches the application commands now that the application_command_create event is obsolete. Maybe we can add a helper method to the module base for fetching the command